File: calc.rs
#!/usr/bin/env -S sh -c 'o=$(mktemp); rustc -o $o $0 && $o'

fn main() {
	std::io::stdin().lines()
		.filter_map(Result::ok)
		.for_each(|l| match l.split(" ").try_fold(Vec::new(), |mut stack, e| {
			if let Ok(n) = e.parse::<i64>() {
				stack.push(n);
				return Ok(stack);
			}

			let b = stack.pop().ok_or("Too Few Arguments")?;
			let a = stack.pop().ok_or("Too Few Arguments")?;

			stack.push(match e {
				"+" => a + b,
				"-" => a - b,
				"*" => a * b,
				"/" => a / b,
				_   => return Err("Invalid Token"),
			});

			Ok(stack)
		}) {
			Ok(s)  if s.len() == 1 => println!("{}", s[0]),
			Ok(_)  => println!("Invalid Input"),
			Err(e) => println!("{e}"),
		})
}

File: calc.pl
#!/usr/bin/env perl

use Scalar::Util qw(looks_like_number);

while (1) {
	chomp(my $line = <STDIN>);
	my @stack;

	foreach (split ' ', $line) {
		(push(@stack, 0 + $_), next) if looks_like_number $_;
		@stack >= 2 or (print("Too Few Arguments\n"), last);

		my ($b, $a) = splice @stack, -2;
		push @stack, $_ eq '+' ? $a + $b
			: $_ eq '-' ? $a - $b
			: $_ eq '*' ? $a * $b
			: $_ eq '/' ? int($a / $b)
			: (print("Invalid Token\n"), last);
	}

	@stack == 1 or (print("Invalid Input\n"), next);
	print $stack[0], "\n";
}
